beacon_chain/state/state_transition.py

```diff
@@ -24,454 +10,321 @@
-            get_incremented_validator_sets(crystallized_state, new_validator_records)
-    else:
-        last_finalized_epoch = crystallized_state.last_finalized_epoch
-        dynasty = crystallized_state.dynasty
-        new_queued_validators = crystallized_state.queued_validators
-        new_active_validators = crystallized_state.active_validators
-        new_exited_validators = crystallized_state.exited_validators
-
-    crystallized_state = CrystallizedState(
-        queued_validators=new_queued_validators,
-        active_validators=new_active_validators,
-        exited_validators=new_exited_validators,
-        current_shuffling=get_shuffling(
-            active_state.randao,
-            len(new_active_validators),
-            config=config
-        ),
-        last_justified_epoch=last_justified_epoch,
-        last_finalized_epoch=last_finalized_epoch,
-        dynasty=dynasty,
-        next_shard=0,
-        current_epoch=crystallized_state.current_epoch + 1,
-        crosslink_records=new_crosslink_records,
-        total_deposits=total_deposits
-    )
-    # Reset the active state
-    active_state = ActiveState(
-        height=active_state.height,
-        randao=active_state.randao,
-        ffg_voter_bitfield=get_empty_bitfield(crystallized_state.num_active_validators),
-        balance_deltas=[],
-        partial_crosslinks=[],
-        total_skip_count=active_state.total_skip_count,
-        recent_proposers=[]
+
+    new_attestations = active_state.pending_attestations + block.attestations
+
+    new_active_state = ActiveState(
+        pending_attestations=new_attestations,
+        recent_block_hashes=active_state.recent_block_hashes[:],
+        block_vote_cache=new_block_vote_cache
     )
+    return new_active_state
 
-    return crystallized_state, active_state
 
+def process_updated_crosslinks(crystallized_state,
+                                active_state,
+                                config=DEFAULT_CONFIG):
+    attestation_votes = {}
+    crosslinks = [c for c in crystallized_state.crosslink_records]
+
+    for attestation in active_state.pending_attestations:
+        shard_tuple = (attestation.shard_id, attestation.shard_block_hash)
+        if shard_tuple not in attestation_votes:
+            attestation_votes[shard_tuple] = 0
 
-def _compute_new_active_state(crystallized_state,
-                              active_state,
-                              parent_block,
-                              block,
-                              verify_sig=True,
-                              config=DEFAULT_CONFIG):
-    # Determine who the attesters and the main signer are
-    attestation_indices, proposer = get_attesters_and_proposer(
+        attestation_indices = get_attestation_indices(
+            crystallized_state,
+            attestation,
+            config
+        )
+        # find total committee size by balance
+        committee_size = sum([
```

Comment:
"committee_size" is used for the number of validators of a committee. Maybe rename to `committee_total_balance`?